src/server/src/main.rs

```diff
@@ -1,23 +1,215 @@
+use bastion::prelude::*;
 use ntex::web;
+use ntex::web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
 
-#[web::get("/")]
-async fn index() -> impl web::Responder {
-    "Hello, World!"
+// user
+#[derive(Deserialize, Serialize, Clone)]
+struct User 
+{
+    username: String,
+    password: String,
+    email: Option<String>,
+    age: Option<u32>,
+    gender: Option<String>,
 }
+// sessionId -> child process reference
+type SessionMap = Arc<Mutex<HashMap<String, ChildRef>>>;
+// username -> User
+type UserMap = Arc<Mutex<HashMap<String, User>>>;
 
-#[web::get("/{name}")]
-async fn hello(name: web::types::Path<String>) -> impl web::Responder {
-    web::HttpResponse::Ok()
-        .content_type("text/plain")
-        //THIS IS FOR THE DEMO, DO NOT DO THIS IN PRODUCTION GET RID OF THIS SOON
-        .set_header(ntex::http::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
-        .body(format!("Hello {}!", &name))
+fn spawn_user_session(username: String) -> ChildRef 
```

Comment:
it shouldn't be possible to create a user session without them being authenticated, so i would like to see the authentication logic moved into here.